core/test/matrix/batch_csr.cpp

```diff
@@ -0,0 +1,511 @@
+    ASSERT_EQ(m->get_num_stored_elements(), 2);
+    EXPECT_EQ(m->get_values()[0], value_type{-1.0});
+    EXPECT_EQ(m->get_values()[1], value_type{-1.0});
+    EXPECT_EQ(m->get_col_idxs()[0], index_type{0});
+    EXPECT_EQ(m->get_row_ptrs()[0], index_type{0});
+    EXPECT_EQ(m->get_row_ptrs()[1], index_type{0});
+    EXPECT_EQ(m->get_row_ptrs()[2], index_type{1});
+}
+
+
+TYPED_TEST(Csr, CanBeDoubleListConstructed)
+{
+    using value_type = typename TestFixture::value_type;
+    using index_type = typename TestFixture::index_type;
+    using BatchCsrMtx = typename TestFixture::BatchCsrMtx;
+    using T = value_type;
+
+    auto m = gko::batch::initialize<BatchCsrMtx>(
+        // clang-format off
+        {{I<T>{1.0, 0.0, 0.0},
+          I<T>{2.0, 0.0, 3.0},
+          I<T>{3.0, 6.0, 0.0}},
+         {I<T>{1.0, 0.0, 0.0},
+          I<T>{3.0, 0.0, -2.0},
+          I<T>{5.0, 8.0, 0.0}}},
+        // clang-format on
+        this->exec, 5);
+
+    ASSERT_EQ(m->get_num_batch_items(), 2);
+    ASSERT_EQ(m->get_common_size(), gko::dim<2>(3, 3));
+    ASSERT_EQ(m->get_num_stored_elements(), 2 * 5);
+    EXPECT_EQ(m->get_values()[0], value_type{1.0});
+    EXPECT_EQ(m->get_values()[1], value_type{2.0});
+    EXPECT_EQ(m->get_values()[2], value_type{3.0});
+    EXPECT_EQ(m->get_values()[3], value_type{3.0});
+    EXPECT_EQ(m->get_values()[4], value_type{6.0});
+    EXPECT_EQ(m->get_values()[5], value_type{1.0});
+    EXPECT_EQ(m->get_values()[6], value_type{3.0});
+    EXPECT_EQ(m->get_values()[7], value_type{-2.0});
+    EXPECT_EQ(m->get_values()[8], value_type{5.0});
+    EXPECT_EQ(m->get_values()[9], value_type{8.0});
+    EXPECT_EQ(m->get_col_idxs()[0], index_type{0});
+    EXPECT_EQ(m->get_col_idxs()[1], index_type{0});
+    EXPECT_EQ(m->get_col_idxs()[2], index_type{2});
+    EXPECT_EQ(m->get_col_idxs()[3], index_type{0});
+    EXPECT_EQ(m->get_col_idxs()[4], index_type{1});
+    EXPECT_EQ(m->get_row_ptrs()[0], index_type{0});
+    EXPECT_EQ(m->get_row_ptrs()[1], index_type{1});
+    EXPECT_EQ(m->get_row_ptrs()[2], index_type{3});
+    EXPECT_EQ(m->get_row_ptrs()[3], index_type{5});
+}
+
+
+TYPED_TEST(Csr, CanBeReadFromMatrixData)
+{
+    using value_type = typename TestFixture::value_type;
+    using index_type = typename TestFixture::index_type;
+    using BatchCsrMtx = typename TestFixture::BatchCsrMtx;
+    auto vec_data = std::vector<gko::matrix_data<value_type, index_type>>{};
+    vec_data.emplace_back(gko::matrix_data<value_type, index_type>(
+        {2, 3}, {{0, 0, -1.0}, {1, 1, 2.5}, {1, 2, 3.5}}));
+    vec_data.emplace_back(gko::matrix_data<value_type, index_type>(
+        {2, 3}, {{0, 0, 1.0}, {1, 1, 2.0}, {1, 2, 3.0}}));
+
+    auto m = gko::batch::read<value_type, index_type, BatchCsrMtx>(this->exec,
+                                                                   vec_data, 3);
+
+    this->assert_equal_to_original_sparse_mtx(m.get());
+}
+
+
+TYPED_TEST(Csr, ThrowsForDataWithDifferentNnz)
```

Comment:
This and the next test are already done in batch_ell. And generally, these tests are not matrix type specific, so they should be in their own file.